### PR TITLE
fix: Publish error when no upstream Worker existed

### LINKED ISSUE #1824: 🐛 BUG: api.error.service_not_found when publishing a development worker
### What version of `Wrangler` are you using?

2.1.0

### What operating system are you using?

Linux + Github Actions

### Describe the Bug

It appears `wrangler publish` broke in the recent version. 

The error I'm getting is:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/ACCOUNTID/workers/services/githubrepo-feature-refs-pull-46-merge) failed.

  workers.api.error.service_not_found [code: 10090]
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

I'm getting this from just running the command `npx wrangler@2 publish --name=test123` too.

When this happens the error is being thrown from [this line](https://github.com/cloudflare/wrangler2/compare/wrangler%402.0.29...wrangler%402.1.0#diff-fa9a52007c2b6571335f75c75e00cf783306b826a082fe144ac0337b3bfff9fcR522). I'm trying to understand the difference between a script and a service (and it's API counterparts), but I can't really seem to figure out what I'm doing wrong

### PR DESCRIPTION
**Pull request by JacobMGEvans**
The check for last deployed was breaking when no available Worker was upstream to check.

fixes #1824

### CODE REVIEW

**Review by petebacondarwin — DISMISSED** [review]
Looking forward to seeing a test that prevents this from regressing 🙏 
Otherwise LGTM

File: packages/wrangler/src/publish.ts [hunk 1]
```diff
@@ -220,21 +220,25 @@ export default async function publish(props: Props): Promise<void> {
...
-		)
-			return;
+			if (
+				(await fromDashMessagePrompt(
+					default_environment.script.last_deployed_from
+				)) === false
+			)
+				return;
+		} catch (e) {
+			// ignore errors
```
Comment (threepointone) [inline packages/wrangler/src/publish.ts:240]: this should only be silent if it's the service not found error

**Review by threepointone — COMMENTED** [review]
add a test?

File: packages/wrangler/src/publish.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/publish.ts]: Done.

**Comment by JacobMGEvans** [comment]
Test added.

File: .changeset/cyan-cherries-stare.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+"wrangler": patch
+---
+
+fix: Publish error when no upstream Worker existed
+try/catch handles the check for last deployed source.
```
Comment (rozenmd) [inline .changeset/cyan-cherries-stare.md:6]:
```suggestion
Add a try/catch when checking when the worker was last deployed
```

File: .changeset/cyan-cherries-stare.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+The check breaking when no available Worker was upstream to check,
```
Comment (rozenmd) [inline .changeset/cyan-cherries-stare.md:7]:
```suggestion
The check was failing when a Worker had never been deployed, causing deployments of new Workers to fail.
```

File: .changeset/cyan-cherries-stare.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+"wrangler": patch
+---
+
+fix: Publish error when no upstream Worker existed
+Add a try/catch when checking when the worker was last deployed
+The check was failing when a Worker had never been deployed, causing deployments of new Workers to fail.
+which was causing deployments of new workers to fail.
```
Comment (rozenmd) [inline .changeset/cyan-cherries-stare.md:8]:
```suggestion
```

**Review by rozenmd — APPROVED** [review]

**Pull request merged**